### ISSUE TITLE
Incorrect routing when using generateStaticParams in standalone mode

### ISSUE THREAD

**Issue by leanderiversen** [vercel/next.js#59505]
### Link to the code that reproduces this issue

https://github.com/leanderiversen/open-next-331/

### To Reproduce

1. Run `next build`
2. Run `node .next/standalone/server.js`
3. Open http://localhost:3000/en-gb/first/second/

### Current vs. Expected behavior

I would expect the static route to render, as this is what happens when running next dev. In reality, the dynamic route renders.

### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 23.3.0: Wed Dec 20 21:30:44 PST 2023; root:xnu-10002.81.5~7/RELEASE_ARM64_T6000
Binaries:
  Node: 18.19.0
  npm: 10.2.3
  Yarn: 1.22.19
  pnpm: 8.15.1
Relevant Packages:
  next: 14.1.1-canary.43 // Latest available version is detected (14.1.1-canary.43).
  eslint-config-next: 14.1.1-canary.43
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.3.3
Next.js Config:
  output: standalone
```

### Which area(s) are affected? (Select all that apply)

App Router, Standalone mode (output: "standalone")

### Additional context

I did initially think that this was an OpenNext issue, but have been provided more details confirming that this is an issue with Next.

**Comment by leanderiversen** [comment]
The issue was fixed before in a canary prior to `next@14.0.5-canary.39`, but seems to have reappeared in a canary between 39 and 57.

**Comment by leanderiversen** [comment]
This looks to be resolved in `14.1.1-canary.15` by [this commit](https://github.com/vercel/next.js/commit/c9321c72c9902bf19bb5c30640d35996a51e56d9).

**Comment by leanderiversen** [comment]
This has now been broken since 14.1.1-canary.16, caused by #61241.
